src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -1113,6 +1116,79 @@ bool isLambdaConversion(BoundExpression expr)
             { WasCompilerGenerated = true };
         }
 
+        private BoundExpression CreateParamsSpan(
+            SyntaxNode syntax,
+            TypeSymbol paramArrayType,
+            ImmutableArray<BoundExpression> arrayArgs)
+        {
+            var elementType = paramArrayType.GetParamsElementType().Type;
+            int length = arrayArgs.Length;
+
+            // PROTOTYPE: If we actually allocate from the stack, it may be necessary to
+            // clear the execution stack before allocation. (See corresponding requirement
+            // for 'stackalloc' in LocalRewriter.VisitStackAllocArrayCreationBase().)
+
+            // For now, simply allocate the array on the heap: new Span<T>(new T[length])
+            var intType = _compilation.GetSpecialType(SpecialType.System_Int32);
+            var array = new BoundArrayCreation(
```

Comment:
>BoundArrayCreation [](http://example.com/codeflow?start=28&length=18)

It looks like we could use ```CreateParamArrayArgument``` helper to create this array.

Reply:
`CreateParamArrayArgument()` creates an array and initializes the values. Ideally, the `Span<T>` here will be created through a BCL helper method so initialization will likely be handled separately from allocation.

Reply:
> the Span<T> here will be created through a BCL helper method so initialization will likely be handled separately from allocation.

When and if this happens, then we will adjust implementation accordingly. At the moment, however, I see no good reason adding all this complexity when we have a perfectly good reusable helper.